Source/WebCore/platform/graphics/haiku/HaikuUtilities.cpp

```diff
@@ -28,44 +28,16 @@
 
 #if PLATFORM(HAIKU)
 
-#include "Logging.h"
-
-#include <Bitmap.h>
-#include <optional>
-#include <OS.h>
-#include <wtf/persistence/PersistentDecoder.h>
-#include <wtf/text/Base64.h>
-#include <wtf/text/WTFString.h>
+#include "NotImplemented.h"
 
 namespace WebCore{
 
 uintptr_t getBitmapUniqueID(BBitmap* bitmap)
 {
     // Gets a unique identifier to a bitmap that is same across processes.
-    // This version gets the id from the id attached to the image's area's name
-    // by SharedMemory::allocate
-
-    area_id area = bitmap->Area();
-
-    area_info info;
-    status_t status = get_area_info(area, &info);
-
-    std::optional<uint64_t> decodedID;
-    if (status == B_OK) {
-        auto base64Data = WTF::String::fromLatin1(&info.name[7]);
-        auto data = WTF::base64Decode(base64Data);
-        WTF::Persistence::Decoder decoder({ data->data(), data->size() });
-        decoder >> decodedID;
-    }
-
-    if (decodedID) {
-        // FIXME: This code is untested. This can be remove once tested.
-        LOG(Compositing, "Found image ID %ld", *decodedID);
-        return *decodedID;
-    } else {
-        LOG(Compositing, "Failed to find image id for BBitmap. Falling back to 0");
-        return 0;
-    }
+    debugger("getBitmapUniqueID needs to be implemented");
```

Comment:
This could probably store the area id and the offset into that area in the uintptr_t with some bit shifting.

Something like this:

    return (area_id << 32) | (area_offset & 0xFFFFFFFF);

We probably won't need areas larger than 4GB.

Reply:
Yes, I was thinking of something like that. However, that only works if I don't need to worry about keeping the ids same across processes. It seems like I don't have to atm since it looks like it's only used by WebProcess.

Anyway, probably best to leave the actual implementation of this till it is actually used so that I get a chance at verifying that assumption.

Reply:
If it's not shared between processes, you can simply return a pointer to the BBitmap. But then I would be surprised that there is a specific API for this.